src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionOperatorsTests.cs

```diff
@@ -15928,944 +17841,1207 @@ static void Main()
+                //         heapLocal += stackLocal;  // 3
+                Diagnostic(ErrorCode.ERR_EscapeCall, "heapLocal += stackLocal").WithArguments("Extensions.extension(S3).operator +(in S3, in S3)", "a").WithLocation(51, 9),
+                // (68,9): error CS8347: Cannot use a result of 'Extensions.extension(S4).operator +(scoped in S4, scoped in S4)' in this context because it may expose variables referenced by parameter 'b' outside of their declaration scope
+                //         heapLocal += stackLocal; // 4
+                Diagnostic(ErrorCode.ERR_EscapeCall, "heapLocal += stackLocal").WithArguments("Extensions.extension(S4).operator +(scoped in S4, scoped in S4)", "b").WithLocation(68, 9),
+                // (68,22): error CS8352: Cannot use variable 'stackLocal' in this context because it may expose referenced variables outside of their declaration scope
+                //         heapLocal += stackLocal; // 4
+                Diagnostic(ErrorCode.ERR_EscapeVariable, "stackLocal").WithArguments("stackLocal").WithLocation(68, 22)
+                );
         }
 
         [Fact]
-        public void CompoundAssignment_069_Consumption_OnObject()
+        public void CompoundAssignment_095_RefSafety()
         {
-            var src = $$$"""
-public static class Extensions1
+            var source = """
+public ref struct C
 {
-    extension(object)
+    public void X(C right) {}
+    public C M1(C c, C c1)
     {
-        public static object operator +(object x, object y)
-        {
-            System.Console.Write("operator1");
-            return x;
-        }
+        c += c1;
+        return c;
+    }
+    public C M2(C c, C c1)
+    {
+        c.X(c1);
+        return c;
     }
 }
 
-class Program
+static class Extensions
 {
-    static void Main()
+    extension(ref C left)
     {
-        var s1 = new object();
-        _ = s1 += s1;
+        public void operator +=(C right) {}
     }
 }
 """;
-
-            var comp = CreateCompilation(src, options: TestOptions.DebugExe);
-            CompileAndVerify(comp, expectedOutput: "operator1").VerifyDiagnostics();
+            CreateCompilation([source, CompilerFeatureRequiredAttribute]).VerifyDiagnostics();
         }
 
         [Fact]
-        public void CompoundAssignment_070_Consumption_OnObject()
+        public void CompoundAssignment_096_RefSafety()
         {
-            var src = $$$"""
-public static class Extensions1
+            var source = """
+public ref struct C
 {
-    extension(object x)
+    public void X(C right) {}
+    public C M1(C c, scoped C c1)
     {
-        public void operator +=(object y)
-        {
-            System.Console.Write("operator1");
-        }
+#line 7
+        c += c1;
+        return c;
+    }
+    public C M2(C c, scoped C c1)
+    {
+        c.X(c1);
+        return c;
     }
 }
 
-class Program
+static class Extensions
 {
-    static void Main()
+    extension(ref C left)
```

Comment:
Nit: this test seemingly wants to draw a parallel between the behavior of the `+=` operator and the method `X`. This change should make them effectively the same.

```suggestion
    extension(scoped ref C left)
```

There are other instances of this pattern in the test file. If you agree with the recommendation I'm happy to add comments for the other. But don't want to add them before hand in case you decide not to take the suggestion